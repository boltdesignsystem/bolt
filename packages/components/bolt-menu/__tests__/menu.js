import { render, stopServer } from '../../../testing/testing-helpers';
import schema from '../menu.schema';
const { spacing, title } = schema.properties;

describe('<bolt-menu> Component', () => {
  afterAll(async () => {
    await stopServer();
  }, 100);

  // Basic Usage
  test('basic usage', async () => {
    const results = await render('@bolt-components-menu/menu.twig', {
      title: 'Menu title',
      items: [
        {
          content: 'Menu item 1',
        },
        {
          content: 'Menu item 2',
          url: 'https://pega.com',
        },
        {
          content: 'Menu item 2',
        },
      ],
    });
    expect(results.ok).toBe(true);
    expect(results.html).toMatchSnapshot();
  });

  // Props
  spacing.enum.forEach(async spacingChoice => {
    test(`menu item spacing: ${spacingChoice}`, async () => {
      const results = await render('@bolt-components-menu/menu.twig', {
        spacing: spacingChoice,
        items: [
          {
            content: 'Menu item 1',
          },
          {
            content: 'Menu item 2',
          },
          {
            content: 'Menu item 2',
          },
        ],
      });
      expect(results.ok).toBe(true);
      expect(results.html).toMatchSnapshot();
    });
  });
});
