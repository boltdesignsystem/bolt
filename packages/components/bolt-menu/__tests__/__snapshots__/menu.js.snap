// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<bolt-menu> Component basic usage 1`] = `
<bolt-menu role="menu">
  <div role="presentation"
       class="c-bolt-menu c-bolt-menu--spacing-small"
  >
    <ssr-keep for="bolt-menu"
              class="c-bolt-menu__title c-bolt-menu__title--spacing-small"
    >
      <span slot="title">
        Menu title
      </span>
    </ssr-keep>
    <ssr-keep for="bolt-menu"
              role="presentation"
    >
      <bolt-menu-item role="presentation">
        <bolt-trigger role="menuitem"
                      display="block"
                      no-outline
        >
          <button type="button"
                  class="c-bolt-trigger c-bolt-trigger--display-block c-bolt-trigger--cursor-pointer c-bolt-trigger--outline-none"
                  is="shadow-root"
          >
            <ssr-keep for="bolt-menu-item"
                      role="presentation"
                      class="c-bolt-menu-item"
            >
              Menu item 1
            </ssr-keep>
          </button>
        </bolt-trigger>
      </bolt-menu-item>
      <bolt-menu-item url="https://google.com"
                      role="presentation"
      >
        <bolt-trigger role="menuitem"
                      url="https://google.com"
                      display="block"
                      no-outline
        >
          <a href="https://google.com"
             target="_self"
             class="c-bolt-trigger c-bolt-trigger--display-block c-bolt-trigger--cursor-pointer c-bolt-trigger--outline-none"
             is="shadow-root"
          >
            <ssr-keep for="bolt-menu-item"
                      role="presentation"
                      class="c-bolt-menu-item"
            >
              Menu item 2
            </ssr-keep>
          </a>
        </bolt-trigger>
      </bolt-menu-item>
      <bolt-menu-item role="presentation">
        <bolt-trigger role="menuitem"
                      display="block"
                      no-outline
        >
          <button type="button"
                  class="c-bolt-trigger c-bolt-trigger--display-block c-bolt-trigger--cursor-pointer c-bolt-trigger--outline-none"
                  is="shadow-root"
          >
            <ssr-keep for="bolt-menu-item"
                      role="presentation"
                      class="c-bolt-menu-item"
            >
              Menu item 2
            </ssr-keep>
          </button>
        </bolt-trigger>
      </bolt-menu-item>
    </ssr-keep>
  </div>
</bolt-menu>
`;

exports[`<bolt-menu> Component menu item spacing: medium 1`] = `
<bolt-menu spacing="medium"
           role="menu"
>
  <div role="presentation"
       class="c-bolt-menu c-bolt-menu--spacing-medium"
  >
    <ssr-keep for="bolt-menu"
              role="presentation"
    >
      <bolt-menu-item role="presentation">
        <bolt-trigger role="menuitem"
                      display="block"
                      no-outline
        >
          <button type="button"
                  class="c-bolt-trigger c-bolt-trigger--display-block c-bolt-trigger--cursor-pointer c-bolt-trigger--outline-none"
                  is="shadow-root"
          >
            <ssr-keep for="bolt-menu-item"
                      role="presentation"
                      class="c-bolt-menu-item"
            >
              Menu item 1
            </ssr-keep>
          </button>
        </bolt-trigger>
      </bolt-menu-item>
      <bolt-menu-item role="presentation">
        <bolt-trigger role="menuitem"
                      display="block"
                      no-outline
        >
          <button type="button"
                  class="c-bolt-trigger c-bolt-trigger--display-block c-bolt-trigger--cursor-pointer c-bolt-trigger--outline-none"
                  is="shadow-root"
          >
            <ssr-keep for="bolt-menu-item"
                      role="presentation"
                      class="c-bolt-menu-item"
            >
              Menu item 2
            </ssr-keep>
          </button>
        </bolt-trigger>
      </bolt-menu-item>
      <bolt-menu-item role="presentation">
        <bolt-trigger role="menuitem"
                      display="block"
                      no-outline
        >
          <button type="button"
                  class="c-bolt-trigger c-bolt-trigger--display-block c-bolt-trigger--cursor-pointer c-bolt-trigger--outline-none"
                  is="shadow-root"
          >
            <ssr-keep for="bolt-menu-item"
                      role="presentation"
                      class="c-bolt-menu-item"
            >
              Menu item 2
            </ssr-keep>
          </button>
        </bolt-trigger>
      </bolt-menu-item>
    </ssr-keep>
  </div>
</bolt-menu>
`;

exports[`<bolt-menu> Component menu item spacing: small 1`] = `
<bolt-menu spacing="small"
           role="menu"
>
  <div role="presentation"
       class="c-bolt-menu c-bolt-menu--spacing-small"
  >
    <ssr-keep for="bolt-menu"
              role="presentation"
    >
      <bolt-menu-item role="presentation">
        <bolt-trigger role="menuitem"
                      display="block"
                      no-outline
        >
          <button type="button"
                  class="c-bolt-trigger c-bolt-trigger--display-block c-bolt-trigger--cursor-pointer c-bolt-trigger--outline-none"
                  is="shadow-root"
          >
            <ssr-keep for="bolt-menu-item"
                      role="presentation"
                      class="c-bolt-menu-item"
            >
              Menu item 1
            </ssr-keep>
          </button>
        </bolt-trigger>
      </bolt-menu-item>
      <bolt-menu-item role="presentation">
        <bolt-trigger role="menuitem"
                      display="block"
                      no-outline
        >
          <button type="button"
                  class="c-bolt-trigger c-bolt-trigger--display-block c-bolt-trigger--cursor-pointer c-bolt-trigger--outline-none"
                  is="shadow-root"
          >
            <ssr-keep for="bolt-menu-item"
                      role="presentation"
                      class="c-bolt-menu-item"
            >
              Menu item 2
            </ssr-keep>
          </button>
        </bolt-trigger>
      </bolt-menu-item>
      <bolt-menu-item role="presentation">
        <bolt-trigger role="menuitem"
                      display="block"
                      no-outline
        >
          <button type="button"
                  class="c-bolt-trigger c-bolt-trigger--display-block c-bolt-trigger--cursor-pointer c-bolt-trigger--outline-none"
                  is="shadow-root"
          >
            <ssr-keep for="bolt-menu-item"
                      role="presentation"
                      class="c-bolt-menu-item"
            >
              Menu item 2
            </ssr-keep>
          </button>
        </bolt-trigger>
      </bolt-menu-item>
    </ssr-keep>
  </div>
</bolt-menu>
`;

exports[`<bolt-menu> Component menu item spacing: xsmall 1`] = `
<bolt-menu spacing="xsmall"
           role="menu"
>
  <div role="presentation"
       class="c-bolt-menu c-bolt-menu--spacing-xsmall"
  >
    <ssr-keep for="bolt-menu"
              role="presentation"
    >
      <bolt-menu-item role="presentation">
        <bolt-trigger role="menuitem"
                      display="block"
                      no-outline
        >
          <button type="button"
                  class="c-bolt-trigger c-bolt-trigger--display-block c-bolt-trigger--cursor-pointer c-bolt-trigger--outline-none"
                  is="shadow-root"
          >
            <ssr-keep for="bolt-menu-item"
                      role="presentation"
                      class="c-bolt-menu-item"
            >
              Menu item 1
            </ssr-keep>
          </button>
        </bolt-trigger>
      </bolt-menu-item>
      <bolt-menu-item role="presentation">
        <bolt-trigger role="menuitem"
                      display="block"
                      no-outline
        >
          <button type="button"
                  class="c-bolt-trigger c-bolt-trigger--display-block c-bolt-trigger--cursor-pointer c-bolt-trigger--outline-none"
                  is="shadow-root"
          >
            <ssr-keep for="bolt-menu-item"
                      role="presentation"
                      class="c-bolt-menu-item"
            >
              Menu item 2
            </ssr-keep>
          </button>
        </bolt-trigger>
      </bolt-menu-item>
      <bolt-menu-item role="presentation">
        <bolt-trigger role="menuitem"
                      display="block"
                      no-outline
        >
          <button type="button"
                  class="c-bolt-trigger c-bolt-trigger--display-block c-bolt-trigger--cursor-pointer c-bolt-trigger--outline-none"
                  is="shadow-root"
          >
            <ssr-keep for="bolt-menu-item"
                      role="presentation"
                      class="c-bolt-menu-item"
            >
              Menu item 2
            </ssr-keep>
          </button>
        </bolt-trigger>
      </bolt-menu-item>
    </ssr-keep>
  </div>
</bolt-menu>
`;
